server/services/group.js

```diff
@@ -89,9 +98,42 @@ function addAdmin(groupName, userId) {
     .then(validatedUserId => Group.addAdmin(groupName, validatedUserId));
 }
 
+function scheduleEPRotation(group) {
+  const rotationInterval = group.escalationPolicy.rotationIntervalInDays;
+  const nextRotateDate = buildRotateDate(group.lastRotated, rotationInterval);
+  return new Promise((resolve) => {
+    scheduler.scheduleJob(nextRotateDate, rotateEscalationPolicy.bind(null, group));
+    resolve({ nextRotateDate, group });
+  });
+}
+
+function rotateEscalationPolicy(group) {
+  const subscribers = group.escalationPolicy.subscribers;
+  const newFirstUser = subscribers.pop();
+  subscribers.unshift(newFirstUser);
+  for (let i = 0; i < subscribers.length; i++) {
+    subscribers[i] = subscribers[i].toString();
+  }
+  const groupUpdates = { lastRotated: new Date() };
+  return scheduleEPRotation(group)
+    .then(() => updateGroup(group.name, groupUpdates))
+    .then(() => updateEscalationPolicy(group.name, { subscribers }));
+}
+
+// Internal Helper Function
```

Comment:
I included it to make it clear that it wouldn't be called by any external service